app/app.go

```diff
@@ -561,13 +561,20 @@ func wirePrioritise(ctx context.Context, conf Config, life *lifecycle.Manager, t
 
 // wireRecaster wires the rebroadcaster component to scheduler, sigAgg and broadcaster.
 // This is not done in core.Wire since recaster isn't really part of the official core workflow (yet).
-func wireRecaster(ctx context.Context, eth2Cl eth2wrap.Client, sched core.Scheduler, sigAgg core.SigAgg, broadcaster core.Broadcaster, validators []state.Validator, builderAPI bool) error {
+func wireRecaster(ctx context.Context, eth2Cl eth2wrap.Client, sched core.Scheduler, sigAgg core.SigAgg,
+	broadcaster core.Broadcaster, validators []state.Validator, builderAPI bool,
+	callback func(context.Context, core.Duty, core.PubKey, core.SignedData) error,
+) error {
 	recaster := bcast.NewRecaster()
 
 	sched.SubscribeSlots(recaster.SlotTicked)
 	sigAgg.Subscribe(recaster.Store)
 	recaster.Subscribe(broadcaster.Broadcast)
 
+	if callback != nil {
+		recaster.Subscribe(callback)
```

Comment:
I was more thinking that you can use the existing wired BroadcastCallback being called by braodcaster. Why do you need a callback from recast itself if it already called broadcast and we can use that callback?

Reply:
How can we use existing wired BroadcastCallback? It is called by sigagg component as broadcaster component has no subscribers:
https://github.com/ObolNetwork/charon/blob/23797a010dc3a98f67f953696df1176f5f186793/app/app.go#L494-L496

That's why I utilized the same callback function as a subscriber for the recaster component. Since the pre-generated registrations are already aggregated, they won't be routed through the sigagg component.

Reply:
ah ok, that is fine then